projects/auth0-angular/src/lib/auth.service.spec.ts

```diff
@@ -126,14 +127,65 @@ describe('AuthService', () => {
       });
     });
 
-    it('should get the user if not authenticated', (done) => {
+    it('should not get the user if not authenticated', (done) => {
+      const user = {
+        name: 'Test User',
+      };
+
+      (auth0Client.isAuthenticated as jasmine.Spy).and.resolveTo(false);
+      (auth0Client.getUser as jasmine.Spy).and.resolveTo(user);
+
+      let value;
+      service.user$.subscribe((v) => {
+        value = v;
+      });
+
+      setTimeout(() => {
+        expect(value).toBeUndefined();
+        done();
+      }, 1000);
+    });
+  });
```

Comment:
As per other comments, I'm not sure this test is super useful. With the way the SDK is at the moment, it's not possible to achieve a result where you have a user or not if you're not authenticated, as that observable only emits something when you are authenticated (you can't have a user otherwise - conceptually it doesn't make sense). I think that makes this test hard to parse; at the beginning we're creating a user object then spying on `getUser` but none of that will be called.

What's the reason for adding this test in relation to `idTokenClaims`? Does it fill a gap that we didn't otherwise have?

Reply:
I think it comes from the discussion here https://github.com/auth0/auth0-angular/pull/69/files#discussion_r509113502 where we talked about verifying nothing gets emitted when not authenticated. 

Maybe this is something we should leave out of the current PR as I think it is worth discussing if we want to have the user$ and idTokenClaims$ emit null when not authenticated.

Doing so would result in the following observables:

**User Is Authenticated**
- user$ emits valid user
- idTokenClaims$ emits valid claims

**User logs out locally only**
- user$ emits null
- idTokenClaims$ emits null

Currently, when the user logsout locally, the user$ and idTokenClaims$ do not reflect that, which might be unexpected.

Reply:
`user$` and `idTokenClaims$` both should emit when authenticated (positive) and not emit when not authenticated (negative). This test is testing the negative scenario, the other test above it tests the positive scenario. That was my thinking, but happy for you to remove or change as you see fit. And yes, spying on `getUser` is redundant - I was thinking that even if `getUser` were to emit, `user$` still shouldn't, but I see now this would never happen.

Reply:
In that case let's remove this test and potentially cover it elsewhere.